src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -1796,5 +1805,239 @@ public long GetLastCounterEtag(DocumentsOperationContext context, string collect
+
+                    key = new Slice(keyByte);
+                    return scope;
+                }
+            }
+
+            public IEnumerable<CounterGroupItemMetadata> GetCountersMetadataFrom(DocumentsOperationContext context, long etag, long skip, long take)
+            {
+                var table = new Table(CountersSchema, context.Transaction.InnerTransaction);
+
+                foreach (var result in table.SeekForwardFrom(CountersSchema.FixedSizeIndexes[AllCountersEtagSlice], etag, skip))
+                {
+                    if (take-- <= 0)
+                        yield break;
+
+                    foreach (var item in TableValueToCounterGroupItemMetadata(context, result.Reader))
+                        yield return item;
+                }
+            }
+
+            public IEnumerable<CounterGroupItemMetadata> GetCountersMetadataFrom(DocumentsOperationContext context, string collection, long etag, long skip, long take)
+            {
+                var collectionName = _countersStorage._documentsStorage.GetCollection(collection, throwIfDoesNotExist: false);
+                if (collectionName == null)
+                    yield break;
+
+                var table = _countersStorage.GetCountersTable(context.Transaction.InnerTransaction, collectionName);
+
+                if (table == null)
+                    yield break;
+
+                foreach (var result in table.SeekForwardFrom(CountersSchema.FixedSizeIndexes[CollectionCountersEtagsSlice], etag, skip))
+                {
+                    if (take-- <= 0)
+                        yield break;
+
+                    foreach (var item in TableValueToCounterGroupItemMetadata(context, result.Reader))
+                        yield return item;
+                }
+            }
+
+            public IDisposable ExtractDocumentIdFromKey(DocumentsOperationContext context, Slice key, out Slice documentId)
+            {
+                int sizeOfDocId = 0;
+                for (; sizeOfDocId < key.Size; sizeOfDocId++)
+                {
+                    if (key[sizeOfDocId] == SpecialChars.RecordSeparator)
+                        break;
+                }
+
+                return Slice.External(context.Allocator, key, sizeOfDocId, out documentId);
+            }
+
+            private static IEnumerable<CounterGroupItemMetadata> TableValueToCounterGroupItemMetadata(DocumentsOperationContext context, TableValueReader tvr)
+            {
+                var etag = TableValueToEtag((int)CountersTable.Etag, ref tvr);
+
+                var valuesData = GetCounterValuesData(context, ref tvr);
+                int size = 0;
+
+                if (valuesData != null)
+                {
+                    size = valuesData.Size;
+
+                    if (valuesData.TryGet(Values, out BlittableJsonReaderObject values))
+                    {
+                        size /= values.Count; // just 'estimating'
```

Comment:
Whats the purpose of size? What do we use it for?

Reply:
For stats, we are using this to record the amount of data that was processed

Reply:
Can Count be 0 here?

Reply:
spoke with @aviv86 it is not possible